core/gui/src/app/workspace/component/code-editor-dialog/code-editor.component.ts

```diff
@@ -169,11 +180,208 @@ export class CodeEditorComponent implements AfterViewInit, SafeStyle, OnDestroy
+            if (acceptButton) acceptButton.removeEventListener("click", acceptListener);
+            if (declineButton) declineButton.removeEventListener("click", declineListener);
+            this.showAnnotationSuggestion = false;
+            resolve();
+            console.log("Resolved!");
+          };
+
+          const acceptListener = () => {
+            this.acceptCurrentAnnotation();
+            cleanup();
+          };
+
+          const declineListener = () => {
+            cleanup();
+          };
+          acceptButton = document.querySelector(".accept-button") as HTMLButtonElement;
+          declineButton = document.querySelector(".decline-button") as HTMLButtonElement;
+
+          if (acceptButton && declineButton) {
+            console.log("Buttons found, adding event listeners");
+            //clean the old one for the "add all type annotation"
+            acceptButton.removeEventListener("click", acceptListener);
+            declineButton.removeEventListener("click", declineListener);
+
+            acceptButton.addEventListener("click", acceptListener, { once: true });
+            declineButton.addEventListener("click", declineListener, { once: true });
+          } else {
+            console.error("Buttons not found!");
+          }
+        }, 0);
+      });
+    });
+  }
+
+  // Called when the user clicks the "accept" button
+  public acceptCurrentAnnotation(): void {
+    // Avoid accidental calls
+    if (!this.showAnnotationSuggestion || !this.currentRange || !this.currentSuggestion) {
+      return;
+    }
+
+    if (this.currentRange && this.currentSuggestion) {
+      const selection = new monaco.Selection(
+        this.currentRange.startLineNumber,
+        this.currentRange.startColumn,
+        this.currentRange.endLineNumber,
+        this.currentRange.endColumn
+      );
+      this.insertTypeAnnotations(this.editor, selection, this.currentSuggestion);
+    }
+    // close the UI after adding the annotation
+    this.showAnnotationSuggestion = false;
+  }
+
+  // Called when the user clicks the "decline" button
+  public rejectCurrentAnnotation(): void {
+    // Do nothing except for closing the UI
+    this.showAnnotationSuggestion = false;
+  }
+
+  // Add the type annotation into monaco editor
+  private insertTypeAnnotations(
+    editor: monaco.editor.IStandaloneCodeEditor,
+    selection: monaco.Selection,
+    annotations: string
+  ) {
+    const endLineNumber = selection.endLineNumber;
+    const endColumn = selection.endColumn;
+    const range = new monaco.Range(
+      // Insert the content to the end of the selected code
+      endLineNumber,
+      endColumn,
+      endLineNumber,
+      endColumn
+    );
+    const text = `${annotations}`;
+    const op = {
+      range: range,
+      text: text,
+      forceMoveMarkers: true,
+    };
+    editor.executeEdits("add annotation", [op]);
+  }
+
+  // Add all necessary modules for type annotation at the first line of the Python UDF
```

Comment:
please give a more descriptive comment here. what's the purpose of those modules?